core/src/main/scala/org/apache/spark/scheduler/TaskDescription.scala

```diff
@@ -52,7 +55,36 @@ private[spark] class TaskDescription(
     val addedFiles: Map[String, Long],
     val addedJars: Map[String, Long],
     val properties: Properties,
-    val serializedTask: ByteBuffer) {
+    private var serializedTask_ : ByteBuffer,
+    private var task_ : Task[_] = null) extends  Logging {
```

Comment:
I wonder if there's any reason to couple the TaskDescription and serialized task now?  Since they're deserialized at different points, I wonder if it makes sense for LaunchTask to have two arguments: one that's the serialized task and one that's the serialized task description (and then to remove task_ and serializedTask as members of this class).  Otherwise, the logic of when task_ is defined (versus null) and how it's deserialized is somewhat difficult to follow.